crud/link_crud.py

```diff
@@ -0,0 +1,103 @@
+from datetime import datetime
+from typing import Any, Sequence
+
+from sqlalchemy import Column, not_, select
+from sqlalchemy.orm import Session
+
+from models import Campaign, ClicksDate, UTMLink
+
+
+def utm_check_exist(session: Session, new_link: UTMLink) -> bool:
+    stmt = select(UTMLink).where(
+        UTMLink.url == new_link.url,
+        UTMLink.campaign_content == new_link.campaign_content,
+        UTMLink.campaign_source == new_link.campaign_source,
+        UTMLink.campaign_medium == new_link.campaign_medium,
+        UTMLink.campaign_name == new_link.campaign_name,
+        UTMLink.domain == new_link.domain,
+        UTMLink.slug == new_link.slug,
+    )
+    return session.execute(stmt).scalar_one_or_none() is not None
+
+
+def utm_add_link(session: Session, new_link: UTMLink) -> None:
+    session.add(new_link)
+    session.commit()
+
+
+def campaign_add(session: Session, new_campaign: Campaign) -> None:
+    session.add(new_campaign)
```

Comment:
Think about use contextmanager and commit when the context manager is gone. But this is not suitable for all cases